### ISSUE TITLE
Semantic versioning in ServiceRegistryRequestDTO

### ISSUE THREAD

**Issue by emanuelpalm** [arrowhead-f/core-java-spring#194]
The [`ServiceRegistryRequestDTO`](https://github.com/arrowhead-f/core-java-spring/blob/master/core-common/src/main/java/eu/arrowhead/common/dto/shared/ServiceRegistryRequestDTO.java) class, which is what is used when registering a service, allows services to optionally specify a version number.

A single number is not very useful for distinguishing non-breaking and breaking version changes, or such including no API changes but include important bug or security fixes. Even though semantic versioning isn't standardized by W3C or any other standards body, it is ubiquitous and very useful. The standard that does exist can be read [here](https://semver.org/).

I understand that changing the field would be difficult at this point, but with this issue at least we could get the discussion started about it.

**Comment by tsvetlin** [comment]
I completely agree with you. All versions should be [semver](https://semver.org/) compliant, as you've suggested it.

For example:
* 1.0.0 is correct.
* 1.0 is accepted;
* Every other variation is refused.

**Comment by jerkerdelsing** [comment]
This should go with updates of which MetaData shall be mandatory. This a current work in there Roadmap WG
